R/object_usage_linter.R

```diff
@@ -196,7 +182,28 @@ extract_glued_symbols <- function(expr) {
       }
     )
   }
-  ls(envir = glued_symbols, all.names = TRUE)
+  names(glued_symbols)
+}
+
+symbol_extractor <- function(text, envir, data) {
+  parsed_text <- tryCatch(
+    parse(text = text, keep.source = TRUE),
+    error = function(...) NULL,
+    warning = function(...) NULL
+  )
+  if (is.null(parsed_text)) {
+    return("")
+  }
+  parse_data <- utils::getParseData(parsed_text)
+  # covers NULL & NA cases
+  if (nrow(parse_data) == 0L) {
+    return("")
+  }
+  symbols <- parse_data$text[parse_data$token == "SYMBOL"]
```

Comment:
Does this approach differentiate between `x` and `` `x` ``?

Reply:
Hmm I think it doesn't, but neither did the old approach.